src/midi/midi_alsa.cpp

```diff
@@ -182,18 +184,33 @@ void MidiHandler_alsa::Close()
 	}
 }
 
-static alsa_address find_seq_input_port()
+static bool port_name_matches(const std::string &pattern,
+                              snd_seq_client_info_t *client_info,
+                              snd_seq_port_info_t *port_info)
+{
+	if (pattern.empty())
+		return true;
+
+	char port_name[40];
+	safe_sprintf(port_name, "%s - %s", snd_seq_client_info_get_name(client_info),
+	             snd_seq_port_info_get_name(port_info));
```

Comment:
How do we prevent writing past 40-chars?
(Maybe the ALSA calls guarantee output of <= 18 chars each?)

Reply:
No, we have no guarantees at all. I have never seen sequencer and port name longer than ~20-30 chars. I'll up the limit to 80 chars - just in case, but I don't think we need to worry about it too much.